app/controllers/SurveyController.java

```diff
@@ -171,7 +171,7 @@ public Result deleteSurvey(String surveyGuid, String createdOnString) throws Exc
         Survey survey = surveyService.getSurvey(keys);
         verifySurveyIsInStudy(session, studyId, survey);
         
-        surveyService.deleteSurvey(studyId, survey);
+        surveyService.deleteSurvey(session.getUser().getHealthCode(), studyId, survey);
```

Comment:
Modified local index to global index on surveyKey, changed all delete apis so they only take a survey.